src/Workspaces/Core/Portable/Editing/DeclarationModifiers.cs

```diff
@@ -153,6 +153,11 @@ public DeclarationModifiers WithIsWriteOnly(bool isWriteOnly)
             return new DeclarationModifiers(SetFlag(_modifiers, Modifiers.WriteOnly, isWriteOnly));
         }
 
+        public DeclarationModifiers WithIsRef(bool isRef)
+        {
+            return new DeclarationModifiers(SetFlag(_modifiers, Modifiers.Ref, isRef));
```

Comment:
Should we add a new constructor with a corresponding parameter as well?

Reply:
Added a new param to the existing constuctor. Is that enough? Do we really need a new constructor for this modifier?

Reply:
Because of compatibility concerns you might have to follow https://github.com/dotnet/roslyn/blob/master/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md but maybe it's better to wait and ask someone from Roslyn

Reply:
Since this is an internal constructor, i think this shoudl be fine.